common/rpc/interceptor/namespace_validator.go

```diff
@@ -164,6 +164,14 @@ func (ni *NamespaceValidatorInterceptor) setNamespace(
 		if request.Namespace == "" {
 			request.Namespace = namespaceEntry.Name().String()
 		}
+	case *workflowservice.RespondNexusTaskCompletedRequest:
```

Comment:
TBH, I don't think this is needed since I would expect the namespace to always be set on the request but I don't know the historical reasons of why this function exists.

Reply:
That's something for us to look into later then, heh